src/app/(marketing)/profile/_components/EditProfile.tsx

```diff
@@ -0,0 +1,216 @@
+import Image from "next/image"
+import { useState } from "react"
+import { FormProvider, useForm } from "react-hook-form"
+
+export const EditProfile = (profile: Schemas["ProfileUpdate"]) => {
+  const form = useForm<Schemas["ProfileUpdate"]>({
+    resolver: zodResolver(profileSchema),
+    defaultValues: {
+      ...profile,
+    },
+    mode: "onTouched",
+  })
+
+  async function onSubmit(values: Schemas["ProfileUpdate"]) {
+    console.log("edit profile submit", values)
+  }
+
+  const { isUploading, onUpload } = usePinata({
+    control: form.control,
+    fieldToSetOnSuccess: "profileImageUrl",
+    onError: (error) => {
+      toast({
+        variant: "error",
+        title: "Failed to upload file",
+        description: error,
+      })
+    },
+  })
+
+  const [uploadProgress, setUploadProgress] = useState(0)
+  const { isDragActive, getRootProps } = useDropzone({
+    multiple: false,
+    noClick: false,
+    onDrop: (acceptedFiles) => {
+      if (!acceptedFiles[0]) return
+      onUpload({
+        data: [acceptedFiles[0]],
+        onProgress: setUploadProgress,
+      })
+    },
+    onError: (error) => {
+      toast({
+        variant: "error",
+        title: `Failed to upload file`,
+        description: error.message,
+      })
+    },
+  })
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button className="-top-8 absolute right-0" variant="solid">
+          <Pencil weight="bold" />
+          Edit profile
+        </Button>
+      </DialogTrigger>
+      <DialogContent size="lg" className="bg-background">
+        <DialogHeader>
+          <DialogTitle>Edit profile</DialogTitle>
+          <DialogCloseButton />
+        </DialogHeader>
+        <FormProvider {...form}>
+          <form onSubmit={form.handleSubmit(onSubmit)}>
+            <DialogBody>
+              <div className="relative mb-20">
+                <FormField
+                  control={form.control}
+                  name="backgroundImageUrl"
+                  render={({ field }) => (
+                    <FormItem className="relative flex h-32 items-center justify-center overflow-hidden rounded-xl border">
+                      <div className="absolute inset-0 size-full">
+                        {field.value ? (
+                          <Image
+                            src={field.value}
+                            width={144}
+                            height={144}
+                            alt="profile background"
+                          />
+                        ) : (
+                          <div className="size-full bg-purple-800" />
+                        )}
+                      </div>
+                      <div className="relative flex items-center gap-3">
+                        <Button size="icon" variant="ghost">
+                          <ImageIcon weight="bold" size={24} />
+                        </Button>
+                        <Separator orientation="vertical" className="h-6 w-0.5" />
```

Comment:
The separator color is inaccurate, I think we can use an arbitrary value here (maybe a white color with 50% opacity), since this is an "edge case".